unit/util/range.cpp

```diff
@@ -57,6 +57,34 @@ SCENARIO("range tests", "[core][util][range]")
       ++it;
       REQUIRE(it == filtered_range.end());
     }
+    THEN("Drop first 2 elements")
+    {
+      auto range = make_range(list);
+      range.drop(2);
+      auto it = range.begin();
+      REQUIRE(*it == "acdef");
+      range.drop(1);
+      REQUIRE(range.empty());
+    }
+    THEN("Drop first 5 elements")
+    {
+      auto range = make_range(list);
+      range.drop(5);
+      REQUIRE(range.empty());
+    }
+    THEN("Skip first 2 elements")
+    {
+      auto range = make_range(list).skip(2);
```

Comment:
I think it would be better to have a test where you also test the original range to make that indeed `skip` does not modify anything, just generates a copy.